### DISCUSSION TITLE
Message Detection and Response

### DISCUSSION THREAD

**Discussion by Skigamer13** [Rapptz/discord.py#6889] [Q&A]
Hi, does anybody here know if there is code that has a bot scan messages for specific words and if the bot detects the words then it will respond?

**Comment by esmaycat** [comment]
You could use the `in` keyword to check if a string is in another string.

**Reply by Skigamer13** [reply]
Sorry, I don't really get what you mean, but after looking in to it more, I came up with this code. Does this also work for what I'm trying to code?

keyword = "bonk"


@bot.event
async def on_message(ctx):
    message_text = ctx.content.strip().upper()
    if keyword in message_text:
        await ctx.send_message(file=discord.File('DogeBot.jpg').format(keyword)
                               )

**Reply by AbstractUmbra** [reply]
It is generally recommended to try and run your own code - testing it - before reaching out for help.
Did you test this code?

Added note: to avoid naming confusion with the commands extension, the parameter given to `on_message` should be `message` or similar, not `ctx` as this is the short form of [`Context`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Context).

**Reply by Skigamer13** [reply]
yeah, I tested it and it made all of the commands stop working. Also, the reason I changed message to ctx is because that is the parameter that is used in the rest of my code so I am trying to stay consistent with my parameters.

**Reply by AbstractUmbra** [reply]
The reason for the first part is [here](https://discordpy.readthedocs.io/en/latest/faq.html#why-does-on-message-make-my-commands-stop-working). Have a look.

As for the second: that is irrelevant. Functions can receive many different types of object, naming them all the same thing is not wise. Correct naming can prevent confusion later on.

**Reply by Skigamer13** [reply]
Ah ok, thank you for the help.

**Reply by CunningBard** [reply]
you can try

keywords = ["horny", "bonk"]
@bot.event
async def on_message():
msg = message.content
if any( word in msg for word in keywords):
    await.message.channel.send("Go to Horny jail")

the code checks if the message has words from the list keywords
